### PR TITLE
Fix parameter type for the BUFFER parameter of the Voronoi polygons a…

### PR DESCRIPTION
**Pull request by havatv**
…lgorithm - fixes #19294

## Description
Changes the type of parameter for a processing algorithm.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Review by nyalldawson — COMMENTED** [review]
Thanks! Could you also rename the parameter to "buffer region (% of extent)" for clarity?

**Comment by havatv** [comment]
Good idea.  I will do it.

**Comment by havatv** [comment]
Fixed in the latest update.

**Comment by nyalldawson** [comment]
Looks good! The pep8 test will fail here - can you run autopep8 on the file (or run scripts/prepare-commit.sh)

**Comment by nyalldawson** [comment]
Thanks! I'll merge as soon as Travis gives it the green light.

File: python/plugins/processing/algs/qgis/VoronoiPolygons.py [hunk 1]
```diff
@@ -72,11 +72,15 @@ def __init__(self):
...
 
     def initAlgorithm(self, config=None):
-        self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT, self.tr('Input layer'), [QgsProcessing.TypeVectorPoint]))
-        self.addParameter(QgsProcessingParameterDistance(self.BUFFER, self.tr('Buffer region'), parentParameterName=self.INPUT,
-                                                         minValue=0.0, maxValue=9999999999, defaultValue=0.0))
+        self.addParameter(QgsProcessingParameterFeatureSource(
+            self.INPUT, self.tr('Input layer'), [QgsProcessing.TypeVectorPoint]))
+        self.addParameter(
+            QgsProcessingParameterNumber(
+                self.BUFFER, self.tr('Buffer region (% of extent)'), parentParameterName=self.INPUT,
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/VoronoiPolygons.py]: Argh, missed this earlier, but you need to remove the parentParameterName argument. That doesn't exist for number parameters.

Comment (havatv) [reply python/plugins/processing/algs/qgis/VoronoiPolygons.py]: Very sorry about that - I should have checked.  I have now removed it.

**Comment by nyalldawson** [comment]
Thank you!

**Pull request merged**